src/init_interp/initInterpMultilevelInterp.F90

```diff
@@ -546,18 +578,41 @@ subroutine interp_onelevel(data_dest, coordinate_dest, data_source, coordinates_
             errMsg(sourcefile, __LINE__))
     end if
 
+    ! Assemble the weights for scaling certain variables by soil thicknesses
+    if (scale_by_thickness) then
+       if (dzsoi_source(index_lower) == 0._r8) then
+          call endrun(msg=subname//' ERROR case 1: about to divide by 0 '// &
+               errMsg(sourcefile, __LINE__))
+       else
+          wt_lower = dzsoi_dest / dzsoi_source(index_lower)
+       end if
+       if (.not. copylevel) then
+          if (dzsoi_source(index_lower+1) == 0._r8) then
+             call endrun(msg=subname//' ERROR case 2: about to divide by 0 '// &
+                  errMsg(sourcefile, __LINE__))
+          else
+             wt_lower_plus_1 = dzsoi_dest / dzsoi_source(index_lower+1)
+          end if
+       else  ! if (copylevel) then wt_lower_plus_1 set but not used
+          wt_lower_plus_1 = wt_lower
+       end if
+    else  ! no scaling; scale factors equal 1
+       wt_lower = 1.0_r8
+       wt_lower_plus_1 = wt_lower
+    end if
+
     ! ------------------------------------------------------------------------
     ! Do the interpolation
     ! ------------------------------------------------------------------------
 
     if ( copylevel) then
-       data_dest = data_source(index_lower)
+       data_dest = data_source(index_lower) * wt_lower
     else
        data_dest = &
-            data_source(index_lower+1) &
+            data_source(index_lower+1) * wt_lower_plus_1 &
             * (coordinate_dest - coordinates_source(index_lower)) &
             / (coordinates_source(index_lower+1) - coordinates_source(index_lower)) + &
-            data_source(index_lower) &
+            data_source(index_lower) * wt_lower &
             * (coordinates_source(index_lower+1) - coordinate_dest ) &
             / (coordinates_source(index_lower+1) - coordinates_source(index_lower))
```

Comment:
@billsacks I wanted to discuss the calculation of `data_dest` here (new lines 609-617). Pls ignore my `wt_lower*` coefficients for now.

As I understand it, we calculate the distances of coord_dest from the bracketing coord_source values, i.e. the distance above and the distance below. We weight the data by these distances and divide by the total distance.

I am confused by the fact that we weight `data_source(index_lower+1)` by the distance from `index_lower` and `data_source(index_lower)` by the distance from `index_lower+1`. It looks backwards to me, though I may have misunderstood. Does it seem correct to you as written?

Reply:
In other words, I would expect
data_dest = (data1 * dist1 + data2 * dist2) / (dist1 + dist2)
where I intentionally refer to distances (calculated from differences in z) rather than thicknesses (that equal dzsoi, which was not available previously).

Reply:
I'm sorry it has taken me so long to get back to you on this. It's possible that I'm still thinking about it wrong, but I think the current implementation is correct. If you consider a destination point really close to source point 1 (say, dist1 = 0.01 and dist2 = 0.99), then you want to give a big weight to source point 1 and a small weight to source point 2. That is, you want to weight source point 1 by the distance from point 2, and vice versa. I think the implementation here follows the standard [linear interpolation between two points](https://en.wikipedia.org/wiki/Linear_interpolation#Linear_interpolation_between_two_known_points). If this still seems wrong to you, let's discuss.

Reply:
@billsacks my intuition was throwing me off. I made a [diagram ](https://docs.google.com/presentation/d/1ka1QkXqZcekQqCmj5S1eMGaEgM-pxZy5jwvvnNoasj4/edit?usp=sharing) that shows why your formula is correct, as well as why my intuition was backwards.

Reply:
Thanks for confirming!